src/pages/Admin/Charity/Permissions/Permissions.tsx

```diff
@@ -0,0 +1,83 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { FormProvider, useForm } from "react-hook-form";
+import { EndowmentController, SettingsPermissions } from "types/contracts";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useEndowmentControllerQuery } from "services/juno/settingsController";
+import { useGetWallet } from "contexts/WalletContext";
+import QueryLoader from "components/QueryLoader";
+import Form from "./Form";
+import { FormField, FormValues, UpdateableFormValues, schema } from "./schema";
+
+export default function Permissions() {
+  const { id } = useAdminResources();
+  const queryState = useEndowmentControllerQuery({ id });
+
+  return (
+    <div className="grid gap-6">
+      <h2 className="font-bold text-3xl">Permissions</h2>
+
+      <QueryLoader
+        queryState={queryState}
+        messages={{
+          error: "Failed to get permissions.",
+          loading: "Loading permissions...",
+        }}
+      >
+        {(endowmentController) => (
+          <InnerComponent controller={endowmentController} />
+        )}
+      </QueryLoader>
+    </div>
+  );
+}
+
+function InnerComponent({ controller }: { controller: EndowmentController }) {
+  const { wallet } = useGetWallet();
+
+  const initialValues: UpdateableFormValues = {
+    accountFees: createField("Changes to account fees", controller.aum_fee),
+    beneficiaries_allowlist: createField(
+      "Changes to beneficiaries whitelist",
+      controller.beneficiaries_allowlist
+    ),
+    contributors_allowlist: createField(
+      "Changes to contributors whitelist",
+      controller.contributors_allowlist
+    ),
+    donationSplitParams: createField(
+      "Changes to donation split parameters",
+      controller.split_to_liquid
+    ),
+    profile: createField("Changes to profile", controller.name),
+  };
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      initialValues,
+      userDelegate:
+        controller.endowment_controller.delegate &&
```

Comment:
```suggestion
        controller.endowment_controller?.delegate?.address  === walllet?.address
```

Reply:
Consider a case where `controller.endowment_controller.delegate === undefined` and the user's wallet is not connected to the webapp, i.e. `wallet?.address === undefined` -> this would make `(controller.endowment_controller?.delegate?.address  === wallet?.address) === true` thus making `userDelegate === true`, enabling unconnected users to make changes to the settings UI (the changes would still not pass due to backend validation).

Reply:
that's right, resolving